lib/OnyxCache.js

```diff
@@ -121,7 +121,7 @@ class OnyxCache {
 
         // lodash adds a small overhead so we don't use it here
         // eslint-disable-next-line prefer-object-spread, rulesdir/prefer-underscore-method
-        this.storageMap = Object.assign({}, utils.fastMerge(this.storageMap, data));
+        this.storageMap = Object.assign({}, utils.fastMerge(this.storageMap, data, false));
```

Comment:
Does this mean we would just leave in the storage map a bunch of keys with `null` and never delete them? I think it does. 

It is a bit hard to wrap one's head around what the side effects of this change would be 😄 

One possible problem is here:

https://github.com/Expensify/react-native-onyx/blob/cd54318794c1682ffb8aee650e9a7983fbceb6f8/lib/Onyx.js#L125-L129

Since a value that we are setting again after deleting previously would be `undefined` but now would be regarded as `null` so we would not check to see if it should be read from storage.

Reply:
We will have more null values in cache storage yes, but this is only the in memory cache and not the persisted storage so I think it is fine to speedup lookup of values that we know do not exist in storage.

The behavior for deleted values should not change since the cache has a drop method which should be used to remove values completely. The only case I know of that we remove values from cache so they will be fetched from storage again is https://github.com/Expensify/react-native-onyx/blob/cd54318794c1682ffb8aee650e9a7983fbceb6f8/lib/OnyxCache.js#L185 which doesn't use merge. In other cases if we use merge with nulls to signal values were deleted it is fine to have them in cache still since we know their latest value is that they do not exist.

I hope that clears up any confusion.

Reply:
>it is fine to speedup lookup of values that we know do not exist in storage.

That's a good point - I was thinking more along the lines of whether this could lead to anything unexpected e.g. could it somehow be possible for the value to exist in storage, but not the cache? But doesn't seem likely.